### PR TITLE
Fix GCP tracking.

### PR DESCRIPTION
**Pull request by zutigrm**
## Summary

Addresses issue:

- #9039

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/legacy-setup/SetupUsingGCP.js [hunk 1]
```diff
@@ -189,32 +199,8 @@ class SetupUsingGCP extends Component {
 		return '';
 	}
 
-	async onButtonClick() {
```
Comment (zutigrm) [inline assets/js/components/legacy-setup/SetupUsingGCP.js:192]: This is actually not invoked, the other button was called from `authentication` step in setup wizard, so I moved the logic fir user setup there, and site setup to other step, once the site is connected

File: assets/js/components/legacy-setup/wizard-step-authentication.js [hunk 2]
```diff
@@ -33,10 +33,48 @@ import { Component } from '@wordpress/element';
...
+
+	async onButtonClick() {
+		const { connectURL, isSiteKitConnected } = this.props;
+
+		await Promise.all( [
+			// Cache the start of the user setup journey.
+			// This will be used for event tracking logic after successful setup.
+			setItem( 'start_user_setup', true ),
+			trackEvent(
+				VIEW_CONTEXT_SPLASH,
```
Comment (aaemnnosttv) [inline assets/js/components/legacy-setup/wizard-step-authentication.js:54]: I just noticed that we're using a different category for the start events than the complete. I'll fix this to use the same for consistency.

**Review by aaemnnosttv — COMMENTED** [review]
A few last things which I'll address quickly now.

**Review by aaemnnosttv — APPROVED** [review]
Great!

**Pull request merged**